### PR TITLE
Sharing Redesign: Export as JSON add toast message

### LINKED ISSUE #90018: Sharing Redesign: Export as JSON add toast message
**Why is this needed**:

For the new Export as JSON drawer, we should add a toast message when it successfully downloads the JSON to keep it consistent with Export as PDF download functionality

**What would you like to be added**:

Add a successful toast message when JSON is downloaded and auto close the drawer

**Who is this feature for?**

Users

### PR DESCRIPTION
**Pull request by lucychen-grafana**
**What is this feature?**

For the new Export as JSON drawer, we should add a toast message when it successfully downloads the JSON to keep it consistent with Export as PDF download functionality

**Why do we need this feature?**
To maintain consistency across download functionality in exporting feature

**Who is this feature for?**

Users

**Which issue(s) does this PR fix?**:

Fixes #https://github.com/grafana/grafana/issues/90018

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx [hunk 1]
```diff
@@ -28,6 +31,13 @@ function ExportAsJsonRenderer({ model }: SceneComponentProps<ExportAsJson>) {
     return JSON.stringify(json, null, 2);
   }, [isSharingExternally]);
 
+  const onClickDownload = async () => {
+    await model.onSaveAsFile();
+    const message = t('export.json.download-successful_toast_message', 'Your JSON has been downloaded');
+    dispatch(notifyApp(createSuccessNotification(message)));
+    dashboard.closeModal();
```
Comment (juanicabanas) [inline public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx:38]: why should we close the drawer? None of the others is closed when an action is fired and it's not the current behavior, neither

Comment (lucychen-grafana) [reply public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx:38]: @natydej has it for export pdf and wanted to extended it to json. But maybe we need to further discuss it

Comment (natydej) [reply public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx]: @lucychen-grafana after discussion with @juanicabanas we consider that closing the drawer in this case is not necessary as in the “save” dashboard drawer, as people may want to select another option after clicking "download", such as copying the JSON file. We should close the drawer only if there is no other action to perform. Therefore, we considered leaving the drawer open for the PDF option as well.

Comment (lucychen-grafana) [reply public/app/features/dashboard-scene/sharing/ExportButton/ExportAsJson.tsx]: @natydej sounds good! But we want to keep the toast message to let them know that download has been successful right?

**Review by juanicabanas — APPROVED** [review]

**Pull request merged**